casper/src/main/scala/coop/rchain/casper/blocks/proposer/Proposer.scala

```diff
@@ -16,11 +16,25 @@ import coop.rchain.casper.util.rholang.RuntimeManager
 import coop.rchain.casper.{Casper, _}
 import coop.rchain.crypto.PrivateKey
 import coop.rchain.metrics.Metrics.Source
+import coop.rchain.metrics.implicits._
 import coop.rchain.metrics.{Metrics, Span}
 import coop.rchain.shared.{EventPublisher, Log, Stopwatch, Time}
 import fs2.Stream
-import coop.rchain.metrics.implicits._
-import coop.rchain.models.BlockHash.BlockHash
+
+sealed abstract class ProposerResult
+object ProposerEmpty                                                         extends ProposerResult
+final case class ProposerSuccess(status: ProposeStatus, block: BlockMessage) extends ProposerResult
+final case class ProposerFailure(status: ProposeStatus, seqNumber: Int)      extends ProposerResult
+final case class ProposerStarted(seqNumber: Int)                             extends ProposerResult
+
+object ProposerResult {
+  def empty: ProposerResult = ProposerEmpty
```

Comment:
I'm curious why `empty`?  and not `busy`

Reply:
@nzpr I was just following what kind of information has this constructor. For busy I'd expect to have info on what is processing like seq number, deploy, current progress, ...